### PR TITLE
Added priority class and leader election to operator helm chart

### PR DESCRIPTION
**Pull request by LP0101**
Added values for priorityClassName and enable-leader-election to the helm chart.

`.Values.config.enableLeaderElection` defaults to true
`.Values.priorityClassName` defaults to `""`

### CODE REVIEW

File: deploy/eck-operator/values.yaml [hunk 1]
```diff
@@ -25,6 +25,8 @@ image:
   # tag is the container image tag. If not defined, defaults to chart appVersion.
   tag: null
 
+priorityClassName: ""
```
Comment (thbkrkr) [inline deploy/eck-operator/values.yaml:28]: Nit: add minimal doc:
```suggestion
# priorityClassName defines the priority of the operator pods
priorityClassName: ""
```

File: deploy/eck-operator/templates/validate-chart.yaml [hunk 2]
```diff
@@ -21,3 +21,9 @@
   {{- fail "Storage class validation cannot be enabled when cluster-scoped resource creation is disabled" -}}
   {{- end -}}
 {{- end -}}
+
+{{- if (not .Values.config.enableLeaderElection) -}}
+  {{- if gt (int .Values.replicaCount) 1 -}}
+  {{- fail "Leader election must be enabled with more than one replica" -}}
+  {{- end -}}
+{{- end -}}
```
Comment (thbkrkr) [inline deploy/eck-operator/templates/validate-chart.yaml:29]: Nit: +empty line in the end of file:
```suggestion
{{- end -}}

```

**Review by thbkrkr — COMMENTED** [review]
Thank you for contributing.

**Comment by thbkrkr** [comment]
Jenkins test this please

**Comment by LP0101** [comment]
Any update on this?

**Review by barkbay — APPROVED** [review]
LGTM

**Comment by barkbay** [comment]
Jenkins test this please

**Pull request merged**